src/tarski/search/model.py

```diff
@@ -28,13 +29,145 @@ class GroundForwardSearchModel:
+        def _is_cost(f):
+            return isinstance(f, tuple) and \
+                'symbol' in dir(f[0]) and \
+                'symbol' in dir(f[0].symbol) and \
+                f[0].symbol.symbol == 'total-cost'
+        return [f for f in state.as_atoms() if not _is_cost(f)]
+
+    def compute_fluents(self):
+        """Compute all of the ground fluents (mentioned in any action, init, goal)"""
+        fluents = set()
+        for op in self.operators:
+            fluents |= set(self._extract_atoms(op.precondition))
+            fluents |= set([e.atom for e in op.effects])
+
+        fluents |= set(self._extract_atoms(self.problem.goal))
+        fluents |= set(self._ignore_cost_metric(self.problem.init))
+
+        return fluents
+
+    def check_strips(self):
+        """Confirms if this operator is a STRIPS operator"""
+        def _strips_condition(c):
+            return isinstance(c, Atom) or \
+                (c.connective == Connective.And and \
+                 all([isinstance(p, Atom) for p in c.subformulas]))
+
+        # Confirm the operators are all STRIPS
+        for op in self.operators:
+            if not _strips_condition(op.precondition):
+                return False
+
+        # Finally, confirm the goal is STRIPS
+        return _strips_condition(self.problem.goal)
+
+    def compute_match_tree(self):
+        """ Compute the match tree for this search model """
+
+        self._id_to_fluent = {idx: f for idx, f in enumerate(self.fluents)}
+        self._fluent_to_id = {f: idx for idx, f in self._id_to_fluent.items()}
+        self._op_to_pre = {}
+        for op in self.operators:
+            self._op_to_pre[op] = {self._fluent_to_id[atom] for atom in self._extract_atoms(op.precondition)}
+
+        # Computes the score of the fluent and partitions the operators accordingly
+        def _score_and_split(ops, fluent):
+            split = {True: [], False: []}
+            for op in ops:
+                split[fluent in self._op_to_pre[op]].append(op)
+            tie_break = {True: 0, False: 0.5}
+            assert len(split[True]) + len(split[False]) == len(ops)
+            return (abs(len(split[True]) - len(split[False])) + tie_break[len(split[True]) > 0], split)
+
+        # Finds the best fluent based on how well it splits the remaining operators
+        def _best_fluent(ops, decisions):
+            best_fluent = None
+            best_score = 999999
+            best_split = None
+            for f in self._id_to_fluent.keys():
+                if f not in decisions:
+                    score, split = _score_and_split(ops, f)
+                    if score < best_score:
+                        best_fluent = f
+                        best_score = score
+                        best_split = split
+            assert best_fluent is not None
+            return (best_fluent, best_split)
+
+        # Recursively computes the match tree
+        def _match_tree(ops, decisions):
+
+            node = {
```

Comment:
It is shame that we cannot use data classes due to compatibility with 3.6, but perhaps using `itertools.namedtuple` will allow you to simplify a bit the code (avoiding to use the verbose and error prone dictionary `['attribute name']` idiom below.